mod/discussions/views/default/resources/discussion/edit.php

```diff
@@ -2,7 +2,7 @@
 
 $guid = elgg_extract('guid', $vars);
 
-elgg_entity_gatekeeper($guid, 'object', 'discussion');
+elgg_entity_gatekeeper($guid, 'object', 'discussion', true);
 
 $topic = get_entity($guid);
 
```

Comment:
i believe the canEdit check can be removed now we have the check in the gatekeeper